### PR TITLE
argK_similarity

### PR DESCRIPTION
**Pull request by arpanbadeka**
created a Kmean file with tika similarity implemented within it.

### CODE REVIEW

**Comment by harsham05** [comment]
@arpanbadeka Why are you modifying vector.py ? There is no need to modify it. Your PR will break cosine similarity metric.

**Comment by arpanbadeka** [comment]
@harsham05 Ohh sorry I just wanted to push argK_similarity file by mistake I made pushed both the files.

**Comment by harsham05** [comment]
Remove your changes to vector.py in your PR. It will break cosine-similarity & k-means metrics

**Comment by chrismattmann** [comment]
thanks @arpanbadeka and @harsham05 once it's removed we can merge this PR. Thanks!

**Comment by harsham05** [comment]
@chrismattmann 

My naive implementation of K-means i.e **argK-means.py** is deprecated & extremely slow. I have a faster implementation committed to Repo.

File: vector.py [hunk 1]
```diff
@@ -84,7 +85,15 @@ def cosTheta(self, v2):
         cosTheta = (V1.V2) / (|V1| |V2|)
         cos 0 = 1 implies identical documents
         '''
-        return self.dotProduct(v2) / (self.getMagnitude() * v2.getMagnitude())
+        #print self.features
```
Comment (chrismattmann) [inline vector.py]: remove commented out

File: vector.py [hunk 2]
```diff
@@ -84,7 +85,15 @@ def cosTheta(self, v2):
...
+        #raw_input()
```
Comment (chrismattmann) [inline vector.py]: remove

File: argK_tika_similarity.py [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+            json_data = {}
+            clusters = []
+            for key in global_minima[1]:    #clusters
+
+                cluster_Dict = {}
+                children = []
+                for point in global_minima[1][key]:
+
+                    node = {}
+                    #fn1 = point.filename.split("\\")[9]
```
Comment (chrismattmann) [inline argK_tika_similarity.py]: remove commented out lines

**Comment by harsham05** [comment]
@chrismattmann This PR is deprecated, due to following

1. Flask based Solr Elasticsearch similarity application is [here](https://github.com/harsham05/Solr-ES-Similarity) which supersedes this PR
2. This PR uses slow & deprecated k-means

**Comment by chrismattmann** [comment]
Dear @harsham05 why don't we merge this first, and then on top of that, create a new PR to pull in your Solr Similarity app and sci-kit based k-means?

**Comment by harsham05** [comment]
Dear @chrismattmann 
There is too much redundant code in tika-similarity and its difficult to proceed forward. I feel tika-similarity should address files for now.

I propose Flask Based Tika Solr Elasticsearch similarity be developed/enhanced separately for now & only address ingestion & visualization of Solr/ES indices

Thank you

**Comment by chrismattmann** [comment]
@harsham05 merging this PR won't hurt, and overall we need to refactor the library anyways. We aren't going to maintain separate repos and I want to credit @arpanbadeka for contributing this. After merging we can work to pull in the external repo you created into tika-similarity and you can give me a demo of how to use it.

**Comment by arpanbadeka** [comment]
Thank you @chrismattmann and @harsham05.

**Pull request merged**